sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py

```diff
@@ -152,27 +191,40 @@ def run_model_and_do_greedy_search(
     device = model.device
     segment_length = server.segment_length
     chunk_length = server.chunk_length
+    decoding_method = server.decoding_method
 
     batch_size = len(stream_list)
 
     state_list = []
-    decoder_out_list = []
-    hyp_list = []
     feature_list = []
+    processed_frames_list = []
+    if decoding_method == "greedy_search":
+        decoder_out_list = []
+        hyp_list = []
+    else:
```

Comment:
Please refactor it in a way that makes it easier to support other decoding methods.